### PR TITLE
Cloudops dockerflow

### PR DESCRIPTION
**Pull request by mostlygeek**
This is a refactored docker build process. After some research CircleCI has better docker support. This PR should be good for review now. This would be the first project to use CircleCI w/ Cloudops Dockerflow. :)

A few things:

- tested it in mostlygeek/browserid-verifier and the docker container build / test / push works
- cloudops will set up https://docker.hub.com/r/mozilla/browserid-verifier repos + circleCI integration
- CircleCI integration settings should be accessible to anybody with permissions to push to this repo

### CODE REVIEW

**Comment by mostlygeek** [comment]
@vladikoff @jrgm please r?

**Comment by mostlygeek** [comment]
Not sure why the travis-ci tests are failing. Seems to be a random bug in the node 0.10 build.

**Comment by vladikoff** [comment]
seemed like it was an npm issue, all fixed now

**Comment by vladikoff** [comment]
LGTM, setting for @jrgm to r+. 

Might be worth to squash the commits into one.

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
+RUN groupadd --gid 1001 app && \
+    useradd --uid 1001 --gid 1001 --home /app --create-home app
+
+WORKDIR /app
+
 
-RUN npm install --unsafe-perm && \
+# Install node requirements and clean up unneeded cache data
+COPY package.json package.json
+RUN su app -c "npm install" && \
```
Comment (jrgm) [inline Dockerfile]: Total nit, but maybe with `su - app -c "npm install"` to clear root's environment variables.

**Comment by jrgm** [comment]
LGTM, although since I haven't worked with circle-ci before, take that for what it's worth.

**Comment by mostlygeek** [comment]
crap i squished the commits the wrong way. :-1:

**Pull request merged**